service/init.go

```diff
@@ -304,6 +315,41 @@ func (pcf *PCF) Exec(c *cli.Context) error {
 	return err
 }
 
+func (pcf *PCF) StartKeepAliveTimer() {
+	var timer *time.Timer
+	//waiting for keepalivetimer value, received after first successful registeration with NRF
+	if val := <-startKeepAliveTimer; val != 0 {
+		logger.InitLog.Infof("Started KeepAlive Timer: %v sec", val)
+		timer = time.NewTimer(time.Duration(val) * time.Second)
+	}
+	for {
+		select {
+		case val := <-startKeepAliveTimer:
+			if timer != nil {
+				timer.Stop()
+			}
+			logger.InitLog.Debugf("Started KeepAlive Timer: %v sec", val)
+			timer = time.NewTimer(time.Duration(val) * time.Second)
+		case <-stopKeepAliveTimer:
+			logger.InitLog.Infof("Stopped KeepAlive Timer")
+			if timer != nil {
+				timer.Stop()
+			}
+		case t := <-timer.C:
+			logger.InitLog.Debugf("KeepAlive timer expired at [%v]", t)
+			nfProfile, err := pcf.updateNF()
+			if err == nil && nfProfile.HeartBeatTimer != 0 {
+				logger.InitLog.Debugf("Started KeepAlive Timer: %v sec", nfProfile.HeartBeatTimer)
+				timer.Reset(time.Duration(nfProfile.HeartBeatTimer) * time.Second)
+			} else {
+				logger.InitLog.Debugf("Started KeepAlive Timer: %v sec", nfProfile.HeartBeatTimer)
```

Comment:
Even in case of error we are coming in else. Its fine, but not sure if you noticed it.